### PR TITLE
gh-285: move test imports to the top

### LINKED ISSUE #285: Import modules globally in test files
We can import libraries and modules globally in the test file. This is more of a style discussion, but mostly it will reduce the number of import calls in test files (DRY + efficient?).

>> I've always put test imports at the top of the file - as have most ARC projects I've worked on. Why put it here?
>
> Other already existing tests follow this pattern and I was trying to maintain uniformity.

_Originally posted by @paddyroddy in https://github.com/glass-dev/glass/pull/263#discussion_r1775263598_

### PR DESCRIPTION
**Pull request by paddyroddy**
Closes #285. Simplifies tests by just moving all imports to the top. I find this easier to read.

Have also simplified the logic to see if `scipy` is present, similarly to `fitsio`.

Also fixed the `broadcast_first` test, which was missing the `test_` prefix.

### CODE REVIEW

**Review by Saransh-cpp — APPROVED** [review]
Thanks, looks good!

File: tests/core/test_array.py [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
+    trapz_product,
+)
 
+# check if scipy is available for testing
+HAVE_SCIPY = importlib.util.find_spec("scipy") is not None
 
-def broadcast_first():
-    from glass.core.array import broadcast_first
 
+def test_broadcast_first():
```
Comment (paddyroddy) [inline tests/core/test_array.py:18]: This would previously not have been run

**Comment by Saransh-cpp** [comment]
Ahh, I need to fix the tests

**Comment by paddyroddy** [comment]
> Ahh, I need to fix the tests

Yeah, and this one isn't working https://github.com/glass-dev/glass/issues/313.

Blocked by #299, #313.

File: tests/core/test_array.py (hunk 1 above)
Comment (paddyroddy) [reply tests/core/test_array.py:18]: #313

**Comment by paddyroddy** [comment]
pre-commit autofix

**Comment by paddyroddy** [comment]
pre-commit.ci autofix

**Comment by paddyroddy** [comment]
Working now, merging

**Pull request merged**